packages/bsky/src/notifications.ts

```diff
@@ -0,0 +1,285 @@
+        .selectFrom('notification_push_token')
+        .where('did', '=', did)
+        .where('token', '=', token)
+        .selectAll()
+        .executeTakeFirst()
+      if (existing) {
+        return
+      }
+
+      // if token doesn't exist, insert it
+      await this.db.db
+        .insertInto('notification_push_token')
+        .values({
+          did,
+          token,
+          platform,
+          endpoint,
+          appId,
+        })
+        .execute()
+    } catch (error) {
+      throw new Error('Failed to insert notification token')
+    }
+  }
+
+  async registerPushNotificationsEndpoint() {}
+
+  async registerPushNotificationsToken() {}
+
+  async getNotificationDisplayAttributes(notif: InsertableNotif) {
+    const {
+      author: authorDid,
+      reason,
+      reasonSubject: subjectUri, // if like/reply/quote/emtion, the post which was liked/replied to/mention is in/or quoted. if custom feed liked, the feed which was liked
+      recordUri,
+    } = notif
+    // 1. Get author's display name
+    const displayName = await this.db.db
+      .selectFrom('profile')
+      .where('creator', '=', authorDid)
+      .select(['displayName'])
+      .executeTakeFirst()
+    if (!displayName || !displayName?.displayName) {
+      throw new Error('Failed to get display name. User has no profile')
+    }
+    const author = NotificationServer.sanitizeDisplayName(
+      displayName.displayName,
+    )
+
+    // 2. Get post data content
+    // if reply, quote, or mention, get URI of the postRecord
+    // if like, or custom feed like, or repost get the URI of the reasonSubject
+    // if follow, get the URI of the author's profile
+    let title = ''
+    let body = ''
+
+    // check follow first and mention first because they don't have subjectUri and return
+    if (reason === 'follow') {
+      title = 'New follower!'
+      body = `${author} has followed you`
+      return { title, body }
+    } else if (reason === 'mention') {
+      title = `${author} mentioned you`
+      const mentionedPostData = await this.db.db
+        .selectFrom('post')
+        .where('uri', '=', recordUri)
+        .selectAll()
+        .executeTakeFirst()
+      body = mentionedPostData?.text || ''
+      return { title, body }
+    }
+
+    if (!subjectUri) {
+      throw new Error('Failed to get subject URI')
+    }
+
+    const postData = await this.db.db
+      .selectFrom('post')
+      .where('uri', '=', subjectUri)
+      .selectAll()
+      .executeTakeFirst()
+    if (!postData) {
+      throw new Error('Failed to get post data')
+    }
+
+    if (reason === 'like') {
+      title = `${author} liked your post`
+      body = postData?.text || ''
+      // custom feed like
+      if (subjectUri?.includes('feed.generator')) {
+        title = `${author} liked your custom feed`
+        body = `${new AtUri(subjectUri).rkey}`
+      }
+    } else if (reason === 'reply') {
+      title = `${author} replied to your post`
```

Comment:
does it make more sense to include the reply text here?

Reply:
Yeah that is probably better. Fixed.